collie/interactions/datasets.py

```diff
@@ -49,6 +50,8 @@ class BaseInteractions(torch.utils.data.Dataset, metaclass=ABCMeta):
     ratings: Iterable[int], 1-d
         Array of corresponding ratings to both ``users`` and ``items``. If ``None``, will default to
         each user in ``user`` interacting with an item with a rating value of 1
+    negative_sample_type: str
+        Type of negative sampling to be performed. One of either ``items`` or ``users``
```

Comment:
I think some more explanation here on what exactly this does. I think this concept of negative user sampling is pretty novel, so it's okay to be pretty verbose here about differences between the two.

Reply:
This is in the docstring, but not in the `__init__` - is that expected?

Reply:
I left it out of the `__init__` because `num_negative_samples` was left out, so I was matching that. Should they both be added to the `__init__`?

Reply:
On second thought, these are both left out of the `__init__` because they are not and should not be inherited by `ExplicitInteractions`. I think both should be removed from this docstring and kept in the `Interactions` docstring, thoughts?

Reply:
Makes sense to me!